### PR TITLE
Fix history view

### PR DESCRIPTION
**Pull request by melnelen**
Simulator Screenshot - iPhone 15 Pro - 2023-11-27 at 12 08 57

### CODE REVIEW

File: FocusOn/View/HistoryView.swift [hunk 1]
```diff
@@ -14,25 +14,56 @@ struct HistoryView: View {
...
-                            HStack {
-                                Text(goal.name)
-                                Spacer()
-                                Image(systemName: (goal.isCompleted ? "checkmark.seal.fill" : "xmark.octagon.fill"))
-                                    .foregroundColor(goal.isCompleted ? Color("SuccessColor") : Color("FailColor"))
-                            }.font(.system(size: 25))
-                            ForEach(Array(goal.tasks), id: \.self) { task in
+                    
+                    // Monthly Summary
+                    ForEach(viewModel.monthlySummaries.sorted(by: { $0.key < $1.key }), id: \.key) { (month, summary) in
```
Comment (cyril-lepottier) [inline FocusOn/View/HistoryView.swift:21]: Ideally, sorting is done by the view model and has unit tests

File: FocusOn/View/HistoryView.swift [hunk 2]
```diff
@@ -14,25 +14,56 @@ struct HistoryView: View {
...
+                                .foregroundColor(Color("AccentColor"))
+                                .padding(.vertical, 5)
+                                .fixedSize(horizontal: false, vertical: true)
+                        }
+        
+                        // Goals for the month
+                        ForEach((viewModel.allGoals?.filter { goal in
+                            let dateFormatter = DateFormatter()
+                            dateFormatter.dateFormat = "MMMM yyyy"
+                            return dateFormatter.string(from: goal.createdAt) == month
```
Comment (cyril-lepottier) [inline FocusOn/View/HistoryView.swift]: Ideally, date formatting is also done by the view model

**Review by cyril-lepottier — APPROVED** [review]

**Comment by melnelen** [comment]
I moved the date formatting and sorting to the view model and added some unit tests

File: FocusOn/View/HistoryView.swift (hunk 1 above)
Comment (cyril-lepottier) [reply FocusOn/View/HistoryView.swift:21]: You should now be able to `remove the sorted(by:)` instruction?

**Review by cyril-lepottier — APPROVED** [review]
It looks much better, to go further if you want to: `fetchGoals() -> [Goal]?` should return a view model object instead of your business object `Goal`. Both will look quite similar, but with some minor differences, for example, one would have the raw date while the other one would have it as an already formatted string.

**Pull request merged**